internal/stopwatch/stopwatch.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package stopwatch
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// Stopwatch is a simple interface to measure time.
+// Stopwatch implementations are not thread-safe.
+type Stopwatch interface {
+	// Start starts the stopwatch and returns the start time.
+	Start() time.Time
+	// Remaining returns the remaining time before the stopwatch expires.
+	Remaining() time.Duration
+	// Expired returns true if the stopwatch has expired.
+	Expired() bool
```

Comment:
```suggestion
	// Exhausted returns true if the stopwatch has exhausted its configured budget.
	Exhausted() bool
```

The documentation may also explain what `Exhausted` means in terms of `Remaining` (is 0) and `Spent`/`Consumed` (is == configured budget)